auth-web/src/store/modules/business.ts

```diff
@@ -40,72 +35,102 @@ export default class BusinessModule extends VuexModule {
-                const approvedName = () => {
-                  for (const nameItem of response.data.names) {
-                    if ([NrState.APPROVED, NrState.CONDITION].includes(nameItem.state)) {
-                      return nameItem.name
-                    }
-                  }
-                }
-
-                BusinessService.updateBusinessName({
-                  businessIdentifier: entity.businessIdentifier,
-                  name: approvedName()
-                })
-
-                // Is a supported IA
-                const isApprovedForIa =
-                  response.data.actions.some(e => e.filingName === LearFilingTypes.INCORPORATION) &&
-                  response.data.state === NrState.APPROVED
-
-                // Conditional State Checks
-                const isConditionallyApproved =
-                  response.data.state === NrState.CONDITIONAL &&
-                  [NrConditionalStates.RECEIVED, NrConditionalStates.WAIVED].includes(response.data.consentFlag)
-
-                // Is approved for one stop registration
-                const isApprovedForRegistration =
-                    response.data.actions.some(e => e.filingName === LearFilingTypes.REGISTRATION)
-
-                entity.nameRequest = {
-                  names: response.data.names,
-                  id: response.data.id,
-                  legalType: response.data.legalType,
-                  nrNumber: response.data.nrNum,
-                  state: response.data.state,
-                  applicantEmail: response.data.applicants?.emailAddress,
-                  applicantPhone: response.data.applicants?.phoneNumber,
-                  enableIncorporation: isApprovedForIa || isConditionallyApproved || isApprovedForRegistration,
-                  folioNumber: response.data.folioNumber,
-                  target: response.data.target
-                }
-              }
-            }).catch(err => {
-              // eslint-disable-next-line no-console
-              console.log(`Error fetching Name Request: ${err}`)
-            })
+    // fetch affiliated entities for this organization
+    const affiliatedEntities = await OrgService.getAffiliatiatedEntities(organization.id)
+      .then(response => {
+        if (response?.data?.entities && response?.status === 200) {
+          return response.data.entities
         }
-      }
+        throw new Error(`Invalid response = ${response}`)
+      })
+      .catch(error => {
+        console.log('Error getting affiliated entities:', error) // eslint-disable-line no-console
+        return [] as Business[]
+      })
+    // console.log('*** affiliated entities =', affiliatedEntities) // eslint-disable-line no-console
 
-      return affiliatedEntities
-    }
+    // get only NR affiliations
+    const nameRequests = affiliatedEntities.map(entity => {
+      return (entity.corpType.code === CorpType.NAME_REQUEST) ? entity : null
+    })
```

Comment:
I did this instead of filtering so that `nameRequests` is the same length as `affiliatedEntities`. I also did this in other statements below. The reason is so that I can write back into affiliatedEntries using an index on line 100.